src/PrestaShopBundle/EventListener/Context/Admin/CountryContextListener.php

```diff
@@ -42,7 +45,7 @@ public function __construct(
 
     public function onKernelRequest(RequestEvent $event): void
     {
-        if (!$event->isMainRequest()) {
+        if (!$event->isMainRequest() && $this->isApiRequest($event->getRequest())) {
```

Comment:
```suggestion
        if (!$event->isMainRequest() || $this->isApiRequest($event->getRequest())) {
```
Isn't it an OR condition here? 🤔 Same comment for all the other listeners since it's the same code